### PR TITLE
[mono] Fix missing simd mul/div case

### PR DESCRIPTION
**Pull request by lewing**
Without this only the scalar cases emit anything for Vector128<T> which is key to emit_vector64_vector128_t working properly. This code was removed in the Vector512 changes probably because it needs additional checks on the types.

### CODE REVIEW

File: src/mono/mono/mini/simd-intrinsics.c [hunk 1]
```diff
@@ -336,6 +339,9 @@ emit_simd_ins_for_binary_op (MonoCompile *cfg, MonoClass *klass, MonoMethodSigna
 						ins = emit_simd_ins (cfg, klass, OP_XBINOP_BYSCALAR, ins->dreg, args [1]->dreg);
 						ins->inst_c0 = OP_FMUL;
 						return ins;
+					} else if ((fsig->params [0]->type == MONO_TYPE_GENERICINST) && (fsig->params [1]->type == MONO_TYPE_GENERICINST)) {
+						instc0 = OP_FMUL;
```
Comment (lewing) [inline src/mono/mono/mini/simd-intrinsics.c:343]:
```suggestion
```

File: src/mono/mono/mini/simd-intrinsics.c [hunk 2]
```diff
@@ -309,6 +309,9 @@ emit_simd_ins_for_binary_op (MonoCompile *cfg, MonoClass *klass, MonoMethodSigna
 						ins = emit_simd_ins (cfg, klass, OP_XBINOP_BYSCALAR, args [0]->dreg, ins->dreg);
 						ins->inst_c0 = OP_FDIV;
 						return ins;
+					} else if ((fsig->params [0]->type == MONO_TYPE_GENERICINST) && (fsig->params [1]->type == MONO_TYPE_GENERICINST)) {
+						instc0 = OP_FMUL;
```
Comment (lewing) [inline src/mono/mono/mini/simd-intrinsics.c:313]:
```suggestion
```

**Comment by lewing** [comment]
https://gist.github.com/radekdoulik/7d27da1693192e16b621f7cd746f2dec has some diffs of good vs bad generated code

**Review by radekdoulik — COMMENTED** [review]
LGTM, it might have been disabled by mistake? I am asking in https://github.com/dotnet/runtime/pull/76642#pullrequestreview-1270709027 to find out if there was a reason to disable it.

**Review by fanyang-mono — APPROVED** [review]
Thanks for fixing this. They were accidentally removed somehow. Seems to be a merge mistake.

**Comment by lambdageek** [comment]
It looks like it was a bad merge into #76642). The correct code is in 8158cb8be51 (from #78784) but then when the PR was merged an incomplete version from #76642was picked instead during the merge (67abfd0)

**Comment by lambdageek** [comment]
I verified this PR restores most of the performance in the threaded raytracer compared to net7

**Review by lambdageek — APPROVED** [review]

**Pull request merged**